src/input/mod.rs

```diff
@@ -249,15 +265,22 @@ impl State {
                                     }
 
                                     // here we can handle global shortcuts and the like
-                                    for (binding, action) in
-                                        data.common.config.static_conf.key_bindings.iter()
-                                    {
-                                        if state == KeyState::Pressed
-                                            && binding.modifiers == *modifiers
-                                            && handle.raw_syms().contains(&binding.key)
+                                    if !shortcuts_inhibited {
+                                        for (binding, action) in
+                                            data.common.config.static_conf.key_bindings.iter()
                                         {
-                                            userdata.get::<SupressedKeys>().unwrap().add(&handle);
-                                            return FilterResult::Intercept(Some(action.clone()));
+                                            if state == KeyState::Pressed
+                                                && binding.modifiers == *modifiers
+                                                && handle.raw_syms().contains(&binding.key)
+                                            {
+                                                userdata
+                                                    .get::<SupressedKeys>()
+                                                    .unwrap()
+                                                    .add(&handle);
+                                                return FilterResult::Intercept(Some(
+                                                    action.clone(),
+                                                ));
+                                            }
                                         }
                                     }
```

Comment:
Maybe we want to have an `else` here in the meantime? Just one complicated (maybe even hard-coded) shortcut to release the lock for the user?

As stated in the protocol:
> The Wayland compositor is however under no obligation to disable all of its shortcuts, and may keep some special key combo for its own use, including but not limited to one allowing the user to forcibly restore normal keyboard events routing in the case of an unwilling client. The compositor may also use the same key combo to reactivate an existing shortcut inhibitor that was previously deactivated on user request.
>  When the compositor restores its own keyboard shortcuts, an "inactive" event is emitted to notify the client that the keyboard shortcuts inhibitor is not effectively active for the surface and seat any more, and the client should not expect to receive all keyboard events.
>
> When the keyboard shortcuts inhibitor is inactive, the client has no way to forcibly reactivate the keyboard shortcuts inhibitor.
> 
> The user can chose to re-enable a previously deactivated keyboard shortcuts inhibitor using any mechanism the compositor may offer, in which case the compositor will send an "active" event to notify the client.

Reply:
Yeah, I was thinking it could have some shortcut that works anyway. I wonder if there is anything that would be obvious to a user trying to get out of a program with shortcuts inhibited... not sure what other compositors are doing.